metaflow/plugins/datatools/s3/s3.py

```diff
@@ -64,17 +68,32 @@ def ensure_unicode(x):
     return None if x is None else to_unicode(x)
 
 
-S3GetObject = namedtuple_with_defaults("S3GetObject", "key offset length")
+PutValue = Union[RawIOBase, BufferedIOBase, str, bytes]
+
+S3GetObject = namedtuple_with_defaults(
+    "S3GetObject", [("key", str), ("offset", int), ("length", int)]
+)
+S3GetObject.__module__ = __name__
 
 S3PutObject = namedtuple_with_defaults(
     "S3PutObject",
-    "key value path content_type metadata",
+    [
+        ("key", str),
+        ("value", Optional[PutValue]),
+        ("path", Optional[str]),
+        ("content_type", Optional[str]),
+        ("metadata", Optional[Dict[str, str]]),
+    ],
     defaults=(None, None, None, None),
 )
+S3PutObject.__module__ = __name__
```

Comment:
what's this for?

Reply:
It was when I was generating the .pyi. I think basically since `namedtuple_with_defaults` is in another module, it was placing those objects there too.